unblob/pool.py

```diff
@@ -82,20 +107,33 @@ def submit(self, args):
         self._input.put(args)
 
     def process_until_done(self):
-        while not self._input.is_empty():
-            result = self._output.get()
-            self._result_callback(self, result)
-            self._input.task_done()
+        try:
+            while not self._input.is_empty():
+                result = self._output.get()
+                self._result_callback(self, result)
+                self._input.task_done()
+        except ShutDownRequired:
+            logger.debug("Shutdown MultiPool.process_until_done")
+            self._should_stop_processing = True
 
 
 class SinglePool(PoolBase):
     def __init__(self, handler, *, result_callback):
         self._handler = handler
         self._result_callback = result_callback
+        self._should_stop_processing = False
```

Comment:
I guess, there is no issue in single processing mode and we don't need to extend this class at all?

Reply:
Just to hide the traces, I guess. But we can leave it out, making the SinglePool cleaner.